### PR TITLE
DDCNL-1200: Updated caching to use for...yield comprehensions for fut…

### PR DESCRIPTION
**Pull request by spattisson**
…ures

### CODE REVIEW

File: test/controllers/MemberDOBControllerSpec.scala [hunk 1]
```diff
@@ -255,10 +256,11 @@ class MemberDOBControllerSpec extends UnitSpec with WithFakeApplication with I18
         )(any())
       }
 
-      "a request is made during the June 2018" in {
+      "a request is made which returns only a CY result (i.e. after 6th April)" in {
 
         when(mockRasConnector.getResidencyStatus(any())(any())).thenReturn(Future.successful(ResidencyStatus(SCOTTISH, None)))
         when(mockSessionService.cacheDob(Matchers.any())(Matchers.any())).thenReturn(Future.successful(Some(rasSession)))
```
Comment (spattisson) [inline test/controllers/MemberDOBControllerSpec.scala]: @nathan-maybrey When I removed this line it goosed the test completely - but it IS a duplicate of line 77 so I don't understand why its OK to leave it in!

File: test/controllers/MemberDOBControllerSpec.scala [hunk 2]
```diff
@@ -231,10 +231,11 @@ class MemberDOBControllerSpec extends UnitSpec with WithFakeApplication with I18
     }
 
     "redirect to match found page" when {
-      "a request is made during the february 2018" in {
+      "a request is made which returns CY and NY results (i.e. before 6th April)" in {
 
         when(mockRasConnector.getResidencyStatus(any())(any())).thenReturn(Future.successful(ResidencyStatus(SCOTTISH, Some(OTHER_UK))))
         when(mockSessionService.cacheDob(Matchers.any())(Matchers.any())).thenReturn(Future.successful(Some(rasSession)))
```
Comment (spattisson) [inline test/controllers/MemberDOBControllerSpec.scala]: ditto (comment below on line 261)

File: app/controllers/RasResidencyCheckerController.scala [hunk 3]
```diff
@@ -50,11 +50,11 @@ trait RasResidencyCheckerController extends RasController {
           None
```
Comment (daniel-manning) [inline app/controllers/RasResidencyCheckerController.scala]: equivalent to rasResponse.nextYearForecastResidencyStatus.map(extractResidencyStatus)

Comment (spattisson) [reply app/controllers/RasResidencyCheckerController.scala]: Added replacement

Comment (spattisson) [reply app/controllers/RasResidencyCheckerController.scala]: Please re-review @daniel-manning

**Pull request merged**